src/silicone/database_crunchers/rms_closest.py

```diff
@@ -189,46 +227,61 @@ def _get_iamdf_section(self, variables):
         return iamdf_section
 
 
-def _select_closest(to_search_df, target_series):
+def _select_closest(to_search_df, target_df, weighting):
     """
-    Find row in ``to_search_df`` that is closest to the target array.
+    Find model/scenario combo in ``to_search_df`` that is closest to that of the target.
 
-    Here, 'closest' is in the root-mean squared sense. In the event that multiple rows
-    are equally close, returns first row.
+    Here, 'closest' is in the root-mean squared sense. In the event that multiple model/
+    scenarios are equally close, returns first row.
 
     Parameters
     ----------
     to_search_df : :obj:`pd.DataFrame`
-        The rows of this dataframe are the candidate closest vectors
+        The model/scenario combos to search for the closest case. A timeseries.
 
-    target_series : :obj:`pd.Series`
-        The vector to which we want to be close
+    target_df : :obj:`pd.DataFrame`
+        The data to which we want to be close. A timeseries.
+
+    weighting : map{str: float}
+        Maps the variable name onto the weighting for that variable.
 
     Returns
     -------
     dict
-        Metadata of the closest row.
+        Index of the closest timeseries.
     """
-    if len(target_series.shape) != 1:
-        raise ValueError("Target array is multidimensional")
 
-    if target_series.shape[0] != to_search_df.shape[1]:
+    if target_df.shape[1] != to_search_df.shape[1]:
         raise ValueError(
             "Target array does not match the size of the searchable arrays"
         )
-
-    closeness = []
+    if any(
+        x not in target_df.index.get_level_values("variable")
+        for x in to_search_df.index.get_level_values("variable")
+    ):
+        raise ValueError("No variable overlap between target and infiller databases.")
+    if any(col not in target_df.columns for col in to_search_df.columns):
+        raise ValueError("Time values mismatch between target and infiller databases.")
+    rms = pd.Series(index=to_search_df.index)
     for label, row in to_search_df.iterrows():
-        rms = (((target_series - row) ** 2).mean()) ** 0.5
-        closeness.append((label, rms))
-
-    # Find the minimum closeness and return the index of it
-    labels, rmss = list(zip(*closeness))
-    to_return = rmss.index(min(rmss))
-    return dict(zip(to_search_df.index.names, labels[to_return]))
+        # The third item in the label is the variable name.
+        rms.loc[label] = (
+            (
+                (
+                    target_df[
+                        target_df.index.get_level_values("variable") == label[3]
+                    ].squeeze()
+                    - row
+                )
+                ** 2
+            ).mean()
+        ) ** 0.5 * weighting[label[3]]
+    rmssums = rms.groupby(level=[0, 1], sort=False).sum()
+    to_return = rmssums.loc[rmssums == min(rmssums)].index.to_list()
```

Comment:
```suggestion
    to_return = rmssums.idxmin()
    # line below then becomes
    return to_return
```

Might be quicker to use inbuilt pandas function? https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.idxmin.html

Reply:
Indeed, and also skip to returning this